### PR TITLE
Relevance helper functions

### LINKED ISSUE #101: Implement two new methods for unit testing of relevance
- [Specs and CTS tests](https://github.com/algolia/algoliasearch-client-specs-internal/pull/41)
- [Go client base implementation](https://github.com/algolia/algoliasearch-client-go/pull/524)

### PR DESCRIPTION
**Pull request by q-litzler**
Fix #101

### CODE REVIEW

File: src/commonTest/kotlin/suite/TestSuiteSearch.kt [hunk 1]
```diff
@@ -46,13 +49,33 @@ internal class TestSuiteSearch {
...
-                search(Query("algolia")).nbHits shouldEqual 2
-                val hits = search(Query("elon", clickAnalytics = true)).hits
 
-                hits.shouldNotBeNull()
-                hits.shouldNotBeEmpty()
+                search(Query("algolia")).apply {
+                    nbHits shouldEqual 2
+                    getObjectIDPosition(ObjectID("nicolas-dessaigne")) shouldEqual 0
+                    getObjectIDPosition(ObjectID("julien-lemoine")) shouldEqual 1
+                    getObjectIDPosition(ObjectID("unknown")) shouldEqual -1
```
Comment (aseure) [inline src/commonTest/kotlin/suite/TestSuiteSearch.kt]: nit: in the CTS, the given string is `""` (empty string) not `"unknown"`

File: src/commonTest/kotlin/suite/TestSuiteSearch.kt [hunk 2]
```diff
@@ -46,13 +49,33 @@ internal class TestSuiteSearch {
...
+                }
+
+                findFirstObject({ false }, Query(""), false).shouldBeNull()
+                findFirstObject({ true }, Query(""), false)!!.apply {
```
Comment (aseure) [inline src/commonTest/kotlin/suite/TestSuiteSearch.kt]: kotlin question: what does `!!` means in that context?

File: src/commonTest/kotlin/suite/TestSuiteSearch.kt (hunk 1 above)
Comment (q-litzler) [reply src/commonTest/kotlin/suite/TestSuiteSearch.kt]: My `ObjectID` class throws if I pass an empty string, is the reason why I put `unknown`

File: src/commonTest/kotlin/suite/TestSuiteSearch.kt (hunk 2 above)
Comment (q-litzler) [reply src/commonTest/kotlin/suite/TestSuiteSearch.kt]: It means throw NullPointerException if the value is null. The test will fail if that's the case.

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt [hunk 3]
```diff
@@ -121,4 +122,26 @@ public interface EndpointSearch {
         filters: Set<Filter>,
         requestOptions: RequestOptions? = null
     ): ResponseSearch
+
+    /**
+     *  Search iteratively through the search response [ResponseSearch.hits] field to find the first response hit tha
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]:
```suggestion
     *  Search iteratively through the search response [ResponseSearch.hits] field to find the first response hit that
```

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt [hunk 4]
```diff
@@ -121,4 +122,26 @@ public interface EndpointSearch {
...
+     *  would match against the give [match] function.
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]:
```suggestion
     *  would match against the given [match] function.
```

File: src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt [hunk 5]
```diff
@@ -121,4 +122,26 @@ public interface EndpointSearch {
...
+     *  If no object has been found within the first result set, the function
+     *  will perform a new search operation on the next page of results, if any,
+     *  until a matching object is found or the end of results, whichever
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/endpoint/EndpointSearch.kt]:
```suggestion
     *  until a matching object is found or the end of results is reached, whichever
```

File: src/commonMain/kotlin/com/algolia/search/model/response/ResponseHitsWithPosition.kt [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package com.algolia.search.model.response
+
+import kotlinx.serialization.Serializable
+
+
+@Serializable
+public data class ResponseHitsWithPosition(
```
Comment (PLNech) [inline src/commonMain/kotlin/com/algolia/search/model/response/ResponseHitsWithPosition.kt]: Single hit => singular :upside_down_face: 

```suggestion
public data class ResponseHitWithPosition(
```

**Pull request merged**